raft-tech/TANF-app
# 2990 - stuck files notification

## Summary of Changes
Pull request closes #2990 

## How to Test

1. Make sure the `SENDGRID_API_KEY` env var is set, reach out if you need a valid key
1. Spin up your environment
    ```
    cd tdrs-backend && docker compose up
    cd tdrs-frontend && docker compose up --build
    ```
1. Sign in and change your user role to `OFA System Admin`
1. Create some submissions. Using `shell_plus`, change the `created_at` to >1hr ago.
1. Reparse some submissions. Using `shell_plus`, change the `timeout_at` to some time in the past. Change `finished` and `success` to `False`.
1. Run `docker compose exec web python manage.py find_pending_submissions`.

Try a lot of combinations. If you can get it to fire in real life stuck-file scenarios, that's a plus.

Example of email
![image](https://github.com/user-attachments/assets/dfa5e965-307d-4d3d-8d7a-c778c11303da)


## Deliverables
_More details on how deliverables herein are assessed included [here](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._

### [Deliverable 1: Accepted Features](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:
+ [x] Notification occurs if a newly submitted file has been in pending state for 1+ hours
+ [x] Notification content matches spec Notification content must include:
    + [x] STT, Program Type, Fiscal Period, Section Name, Submission Date/time
    + [x] Dynamic link that passes data file ID to the django URL string
    + [x] Confirm these email events should be captured in logs
+ [x] All sys-admin users receive the notification

### [Deliverable 2: Tested Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
+ Are code coverage minimums met?

### [Deliverable 3: Properly Styled Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)


### [Deliverable 4: Accessible](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)



### [Deliverable 5: Deployed](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)


### [Deliverable 6: Documented](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)


### [Deliverable 7: Secure](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)


### [Deliverable 8: User Research](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

Research product(s) clearly articulate(s):
